fuzzers/005-tilegrid/cfg/top.py

```diff
@@ -34,6 +34,8 @@ def run():
     jtag_chains = ("1", "2", "3", "4")
     for (tile_name, site_name), isone in zip(sites,
                                              util.gen_fuzz_states(len(sites))):
+        site_name = site_name[:-1]
```

Comment:
This is confusing, can you write a comment explaining why you are doing this?

Reply:
It looks like perhaps the spartan7 requires that the user config parameter and the location of the BSCANE2 block be correlated: that is,

BSCANE2 at X0Y0 => user register 1
BSCANE2 at X0Y1 => user register 2
...

This code here takes the name (BSCANE2_X0Y0), chops off the last digit (which is always 0 in the original code), and adds a sequence number that increments with the USER register parameter, thus creating a correlation of the block location and the user register number. 

Basically, the original code as written would take the same block (BSCANE2 at X0Y0) and assign once to each of the user registers 1-4, this code correlates the four possible positions of the BSCANE2 to one each of the user registers.

Reply:
It's basically a really dumb way to derive the name of the block. Probably a better way would be to figure out how to extract the name from the tile database without having to edit it using string operations, but I couldn't figure that out.

Reply:
This same issue showed up when I ran the existing artix7 fuzzers with a newer version of Vivado. I suspect it may be a result of the Vivado version used, rather than being a spartan vs artix difference.